### DISCUSSION TITLE
Trying to open one telnet connection and display data on different tabs? Can I get there with tabby?

### DISCUSSION THREAD

**Discussion by jnz86** [Eugeny/tabby#9849] [Q&A]
I have a local network device, it provides a telnet connection. Once open, it sends data that is prefaced with a number tag. I'd like to post that data on same numbered tabs.

So 

`1> DATA FOR PAGE1 <1`
`2> DATA FOR PAGE2 <2`
etc

I have three tasks.

1. Make the connection. It is bidirectional but the upload is rare and limited
2. Take the incoming data and split it out to different tabs
3. Display the correct data in the correct tab, hopefully respecting ASCII/terminal/serial color specs.

Can I get there with tabby? I might have to do this via a plugin. Or I might have to write something with no UI that opens the telnet connection, then posts to a a bunch of fake-serial interfaces and in tabby have a tab for each one. 

I'd like this to work so that I open tabby with my script or plugin, connect to localhost:port, and all 1-16 tabs are already open or they auto open as data comes in.

Just looking for suggestions on how to get started. I haven't looked at the plugin system yet.

**Comment by Eugeny** [comment]
Figuring out how to write a plug-in that simulates those connections would probably be an overkill imo.
It's easier to write a simple proxy that listens on 16 ports and sorts and forwards data to connections on those ports. 

Then you add 16 telnet profiles to Tabby to connect to these

**Reply by jnz86** [reply]
Unfortunately my device that is serving the telnet data is doing so as a blackbox. I can only make one telnet connection on one port, and the data it posts is "tagged" with a prefix and sufix. 

So I'll have to have something somewhere parsing those at every newline, and pasting them somewhere. Ideally this would be to a tabby tab. 

Which is why I thought maybe a plugin could do this. I assume a plugin could make a telnet connection, parse data, and paste it to a tab that exists and if the correct one doesn't exist to open a new one. Maybe?
